retail/interactive-tutorial/RetailEvents.Samples/PurgeUserEventSample.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            EventType = "detail-page-view",
+            VisitorId = "test_visitor_id",
+            EventTime = DateTime.UtcNow.ToTimestamp()
+        };
+
+        userEvent.ProductDetails.Add(productDetail);
+
+        Console.WriteLine("User Event:");
+        Console.WriteLine($"Product detail: {userEvent.ProductDetails}");
+        Console.WriteLine($"Event type: {userEvent.EventType}");
+        Console.WriteLine($"Visitor id: {userEvent.VisitorId}");
+        Console.WriteLine($"Event time: {userEvent.EventTime}");
+        Console.WriteLine();
+
+        return userEvent;
+    }
+
+    /// <summary>
+    /// Get the purge user event request.
+    /// </summary>
+    /// <param name="defaultCatalog">The default catalog.</param>
+    /// <returns>The purge user event request.</returns>
+    private static PurgeUserEventsRequest GetPurgeUserEventsRequest(string defaultCatalog)
+    {
+        PurgeUserEventsRequest purgeRequest = new PurgeUserEventsRequest
+        {
+            Parent = defaultCatalog,
+            Filter = $"visitorId=\"test_visitor_id\"", // To check error handling set invalid filter here.
+            Force = true
+        };
+
+        Console.WriteLine("Purge user event. request:");
+        Console.WriteLine($"Parent: {purgeRequest.Parent}");
+        Console.WriteLine($"Filter: {purgeRequest.Filter}");
+        Console.WriteLine($"Force: {purgeRequest.Force}");
+        Console.WriteLine();
+
+        return purgeRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to purge user events.
+    /// </summary>
+    /// <param name="defaultCatalog">The default catalog.</param>
+    /// <returns>The completed purge response.</returns>
+    public static Operation<PurgeUserEventsResponse, PurgeMetadata> CallPurgeUserEvents(string defaultCatalog)
+    {
+        PurgeUserEventsRequest purgeRequest = GetPurgeUserEventsRequest(defaultCatalog);
+        UserEventServiceClient client = UserEventServiceClient.Create();
+        Operation<PurgeUserEventsResponse, PurgeMetadata> purgeResponse = client.PurgeUserEvents(purgeRequest);
+
+        Console.WriteLine("The purge operation was started:");
+        Console.WriteLine(purgeResponse.Name);
+        Console.WriteLine();
+
+        return purgeResponse;
+    }
+}
+// [END retail_purge_user_event]
+
+/// <summary>
+/// The purge user event tutorial class.
+/// </summary>
+public static class PurgeUserEventTutorial
+{
+    [Runner.Attributes.Example]
+    public static void PerformPurgeUserEventOperation()
+    {
+        string projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        string defaultCatalog = $"projects/{projectId}/locations/global/catalogs/default_catalog";
+
+        // To check the error handling try to pass invalid catalog:
+        // defaultCatalog = "projects/{projectId}/locations/global/catalogs/invalid_catalog";
+
+        UserEvent userEventToWrite = PurgeUserEventSample.GetUserEvent();
```

Comment:
Does the WriteUserEventSample doesn't write its own event? Do you need a specific even, and a specific even method here for some reason?
If yes, then just remove it from sample code (i.e. from the region tags) as it is not being used anywhere except here in the tutorial.

Reply:
Removed it from **Purge** and **Rejoin** samples. Now, the method **GerUserEvent()** exists only in **WriteUserEvent** sample.